mm2src/coins/eth.rs

```diff
@@ -1098,6 +1101,44 @@ impl MarketCoinOps for EthCoin {
 
     fn my_address(&self) -> Result<String, String> { Ok(checksum_address(&format!("{:#02x}", self.my_address))) }
 
+    fn get_public_key(&self) -> Result<String, MmError<UnexpectedDerivationMethod>> { unimplemented!() }
+
+    /// Hash message for signature using Ethereum's message signing format.
+    /// keccak256(PREFIX_LENGTH + PREFIX + MESSAGE_LENGTH + MESSAGE)
+    fn sign_message_hash(&self, message: &str) -> Option<[u8; 32]> {
+        let message_prefix = self.sign_message_prefix.as_ref()?;
+        let mut stream = Stream::new();
+        let prefix_len = CompactInteger::from(message_prefix.len());
+        prefix_len.serialize(&mut stream);
+        stream.append_slice(message_prefix.as_bytes());
+        stream.append_slice(message.len().to_string().as_bytes());
+        stream.append_slice(message.as_bytes());
+        Some(keccak256(&stream.out()).take())
+    }
+
+    fn sign_message(&self, message: &str) -> SignatureResult<String> {
+        let message_hash = self.sign_message_hash(message).ok_or(SignatureError::PrefixNotFound)?;
+        let privkey = &self.key_pair.secret();
+        let signature = sign(privkey, &H256::from(message_hash))?;
+        Ok(signature.to_string())
+    }
+
+    fn verify_message(&self, signature: &str, message: &str, address: &str) -> VerificationResult<bool> {
+        let message_hash = self
+            .sign_message_hash(message)
+            .ok_or(VerificationError::PrefixNotFound)?;
+        let address = self
+            .address_from_str(address)
+            .map_err(VerificationError::AddressDecodingError)?;
+        let signature = if &signature[..2] == "0x" {
```

Comment:
Can be single line: `let signature = Signature::from_str(signature.strip_prefix("0x").unwrap_or(signature))?;`.